### DISCUSSION TITLE
Project Scope and possible extension

### DISCUSSION THREAD

**Discussion by hrueger** [NotesHubApp/git-essentials#50] [General]
Hi @alex-titarenko,
I come from [isomorphic-git](https://github.com/isomorphic-git/isomorphic-git) and I'm so glad that this repo exists. I've found the community over there a little bit closed-minded about different things like typescript or the exposure of the internal api.

I'd like to ask how you want this project to evolve. Would you be interested in me contributing some more features that I'm missing right now (e.g. readBlob)? I'm also working on the server side and have a JS-only implementation of `git-upload-pack`.
Are these features that you consider to be in the scope of git-essentials?

Best regards

**Comment by alex-titarenko** [comment]
Hi @hrueger first of all I want to start that this project emerged from needs for NotesHub (https://noteshub.app) application. You can also see that git-essentials is located under NotesHubApp organization. This is explanation for the current set of API, that is minimum required for NotesHub to operate. I don't expect much development on my side here unless it require some changes by NotesHub.

But since this open-source project and it might be used by more folks which may need different set of API, I'm open for contribution here. So you can create PR to add readBlob/writeBlob APIs and if everything looks good, I'll accept it.

Regarding `git-upload-pack` API, this is not commonly used API, so I'll probably not accept it.

As the name of the project stated 'git-essentials' so it contains only essential commands from git to keep it small and clean and well maintained.
